.github/workflows/check_for_new_files.yml

```diff
@@ -0,0 +1,75 @@
+name: Validate CODEOWNERS
+on:
+  pull_request:
+    branches: [main]
+env:
+  DIFF_DIRECTORIES: 'src/pages src/fragments'
+jobs:
+  checkIfCodeownersUpdateNeeded:
+    permissions:
+      pull-requests: write
+    name: Check if CODEOWNERS needs to be updated
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+      - name: Get names of added files
+        env:
+          GITHUB_PULL_REQUEST_HEAD_SHA: ${{ github.pull_request.head.sha }}
+        run: |
+          git fetch origin main
+          echo "NEW_FILES<<EOF" >> $GITHUB_ENV
+          echo $(git diff --name-only --diff-filter=A origin/main -- ${{ env.DIFF_DIRECTORIES }} ${{ env.GITHUB_PULL_REQUEST_HEAD_SHA }}) >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+          echo "NEW_FILES_COUNT<<EOF" >> $GITHUB_ENV
+          echo $(git diff --name-only --diff-filter=A origin/main -- ${{ env.DIFF_DIRECTORIES }} ${{ env.GITHUB_PULL_REQUEST_HEAD_SHA }} | wc -l) >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+      - name: Print out new file names
+        run: |
+          echo "New files: ${{ env.NEW_FILES }}"
+      - name: Install npm package ignore
+        if: ${{ env.NEW_FILES_COUNT > 0 }}
+        run: yarn add ignore@5.2.4 -W # help verify CODEOWNERS
+      - name: Check file against CODEOWNERS
+        if: ${{ env.NEW_FILES_COUNT > 0 }}
+        env:
+          CURRENT_BRANCH: ${{ github.head_ref }}
+        uses: actions/github-script@v6.4.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const ignore = require('ignore');
+            const { NEW_FILES, CURRENT_BRANCH } = process.env
+
+            console.log('New files: ', NEW_FILES);
+
+            const response = await fetch(`https://raw.githubusercontent.com/aws-amplify/docs/${CURRENT_BRANCH}/.github/CODEOWNERS`);
```

Comment:
The goal of this workflow is that if a new file is added and it isn't covered by a CODEOWNER, then the current PR should make an update to specify it. By fetching the CODEOWNERS file from the current branch here we can keep track of when the files do get added to CODEOWNERS.